src/hooks/useSortByTopic.ts

```diff
@@ -0,0 +1,28 @@
+import { useMemo } from 'react'
+import useSafeState from './useSafeState'
+
+interface SortTopic {
+  topic: string
+  order: boolean
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function useSortByTopic<T>(
+  data: T[],
+  defaultTopic: string,
```

Comment:
Could use something like this to avoid passing in a topic that doesn't exist:

```suggestion
function useSortByTopic<T, K extends keyof T>(
  data: T[],
  defaultTopic: K,
```

:warning: Code above was not tested, might require changes.

Reply:
I was suggesting a similar thing in this other comment (https://github.com/gnosis/dex-react/pull/1067#discussion_r431851117), but:

instead of `K extends keyof T` 
I would say `K extends string` 

so `type OrderSortByField = 'validUntil' | 'otherSorting' | ...` would work with it too